clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4777,6 +4782,10 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
     O << "    return 0;\n";
     O << "#endif\n";
     O << "  }\n";
+    O << "  // Returns the size of the kernel object in bytes.\n";
+    O << "  __SYCL_DLL_LOCAL\n";
+    O << "  static constexpr int64_t getKernelSizeof() { return "
```

Comment:
I was ok with `getKernelSize` (commented this was an old suggestion), but let @Fznamznon choose which she prefers.

Reply:
> Also, is int64_t going to be ok here? I dont know what amount of 'includes' we have in the header (since int64_t is a typedef).

Good point.   I will check on both Linux and Windows.

Reply:
> Okay, will wait for @Fznamznon.

It is LGTM now.

Reply:
LIT tests show that Windows is not ok with that, see  buildbot/sycl-win-x64-pr job result . We have a test that compiles integration header.

Reply:
Regarding the name: I changed it to getKernelSize.   getKernelSizeof felt a little weird.

Reply:
> Regarding the name: I changed it to getKernelSize. getKernelSizeof felt a little weird.

SGTM, I don't disagree.